diff --git b/rs/sns_aggregator/src/types/ic_sns_ledger.rs a/rs/sns_aggregator/src/types/ic_sns_ledger.rs
index 7566582df..00c91ff35 100644
--- b/rs/sns_aggregator/src/types/ic_sns_ledger.rs
+++ a/rs/sns_aggregator/src/types/ic_sns_ledger.rs
@@ -129,7 +129,7 @@ pub struct GetBlocksResponseArchivedBlocksItem {
     pub start: BlockIndex,
     pub length: candid::Nat,
 }
-#[derive(Serialize, Clone, Debug, CandidType, Deserialize)]
+#[derive(CandidType, Deserialize)]
 pub struct GetBlocksResponse {
     pub certificate: Option<serde_bytes::ByteBuf>,
     pub first_index: BlockIndex,
@@ -205,7 +205,7 @@ pub struct GetTransactionsResponseArchivedTransactionsItem {
     pub start: TxIndex,
     pub length: candid::Nat,
 }
-#[derive(Serialize, Clone, Debug, CandidType, Deserialize)]
+#[derive(CandidType, Deserialize)]
 pub struct GetTransactionsResponse {
     pub first_index: TxIndex,
     pub log_length: candid::Nat,
@@ -339,7 +339,7 @@ pub struct GetBlocksResultArchivedBlocksItem {
     pub args: Vec<GetBlocksArgs>,
     pub callback: GetBlocksResultArchivedBlocksItemCallback,
 }
-#[derive(Serialize, Clone, Debug, CandidType, Deserialize)]
+#[derive(CandidType, Deserialize)]
 pub struct GetBlocksResult {
     pub log_length: candid::Nat,
     pub blocks: Vec<GetBlocksResultBlocksItem>,
