diff --git b/rs/sns_aggregator/src/types/ic_sns_swap.rs a/rs/sns_aggregator/src/types/ic_sns_swap.rs
index 49318b427..fd6ab7730 100644
--- b/rs/sns_aggregator/src/types/ic_sns_swap.rs
+++ a/rs/sns_aggregator/src/types/ic_sns_swap.rs
@@ -170,7 +170,7 @@ pub struct SettleCommunityFundParticipationResult {
 
 #[derive(Serialize, Clone, Debug, CandidType, Deserialize)]
 pub enum Possibility2 {
-    Ok(EmptyRecord),
+    Ok,
     Err(CanisterCallError),
 }
 
