diff --git b/rs/sns_aggregator/src/types/ic_sns_root.rs a/rs/sns_aggregator/src/types/ic_sns_root.rs
index 32f02ed51..5abd23a7c 100644
--- b/rs/sns_aggregator/src/types/ic_sns_root.rs
+++ a/rs/sns_aggregator/src/types/ic_sns_root.rs
@@ -133,7 +133,7 @@ pub struct GetSnsCanistersSummaryResponse {
 #[derive(CandidType, Deserialize, Serialize, Clone, Debug)]
 pub struct ListSnsCanistersArg {}
 
-#[derive(CandidType, Deserialize, Serialize, Clone, Debug)]
+#[derive(CandidType, Deserialize, Serialize, Clone, Debug, Default)]
 pub struct ListSnsCanistersResponse {
     pub root: Option<candid::Principal>,
     pub swap: Option<candid::Principal>,
