diff --git b/rs/sns_aggregator/src/types/ic_sns_swap.rs a/rs/sns_aggregator/src/types/ic_sns_swap.rs
index d7b6fe69b..60992d269 100644
--- b/rs/sns_aggregator/src/types/ic_sns_swap.rs
+++ a/rs/sns_aggregator/src/types/ic_sns_swap.rs
@@ -172,7 +172,7 @@ pub struct SettleCommunityFundParticipationResult {
 
 #[derive(Serialize, Clone, Debug, CandidType, Deserialize)]
 pub enum Possibility2 {
-    Ok(EmptyRecord),
+    Ok,
     Err(CanisterCallError),
 }
 
