diff --git b/rs/sns_aggregator/src/types/ic_sns_root.rs a/rs/sns_aggregator/src/types/ic_sns_root.rs
index 55c139f45..fbdbe4e10 100644
--- b/rs/sns_aggregator/src/types/ic_sns_root.rs
+++ a/rs/sns_aggregator/src/types/ic_sns_root.rs
@@ -136,7 +136,7 @@ pub struct GetSnsCanistersSummaryResponse {
 #[derive(Serialize, Clone, Debug, CandidType, Deserialize)]
 pub struct ListSnsCanistersArg {}
 
-#[derive(Serialize, Clone, Debug, CandidType, Deserialize)]
+#[derive(Serialize, Clone, Debug, CandidType, Deserialize, Default)]
 pub struct ListSnsCanistersResponse {
     pub root: Option<Principal>,
     pub swap: Option<Principal>,
