diff --git b/rs/sns_aggregator/src/types/ic_sns_wasm.rs a/rs/sns_aggregator/src/types/ic_sns_wasm.rs
index a539e1e2c..ea2adb367 100644
--- b/rs/sns_aggregator/src/types/ic_sns_wasm.rs
+++ a/rs/sns_aggregator/src/types/ic_sns_wasm.rs
@@ -280,7 +280,7 @@ pub struct InsertUpgradePathEntriesResponse {
 #[derive(CandidType, Deserialize, Serialize, Clone, Debug)]
 pub struct ListDeployedSnsesArg {}
 
-#[derive(CandidType, Deserialize, Serialize, Clone, Debug)]
+#[derive(CandidType, Deserialize, Serialize, Clone, Debug, Default)]
 pub struct DeployedSns {
     pub root_canister_id: Option<candid::Principal>,
     pub governance_canister_id: Option<candid::Principal>,
