diff --git b/rs/sns_aggregator/src/types/ic_sns_root.rs a/rs/sns_aggregator/src/types/ic_sns_root.rs
index 1dc8c34d2..2f942ab47 100644
--- b/rs/sns_aggregator/src/types/ic_sns_root.rs
+++ a/rs/sns_aggregator/src/types/ic_sns_root.rs
@@ -126,7 +126,7 @@ pub struct GetSnsCanistersSummaryResponse {
 #[derive(CandidType, Deserialize, Serialize, Clone, Debug)]
 pub struct list_sns_canisters_arg0 {}
 
-#[derive(CandidType, Deserialize, Serialize, Clone, Debug)]
+#[derive(CandidType, Deserialize, Serialize, Clone, Debug, Default)]
 pub struct ListSnsCanistersResponse {
     pub root: Option<candid::Principal>,
     pub swap: Option<candid::Principal>,
